### DISCUSSION TITLE
Automatic ID [Jun 15]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2459] [Discord Threads]
> 🙋‍♂️ **geocoast**

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### when you create a table in nocodb - we generate an id which is set to autoincrement  
#  
> 🙋‍♂️ **geocoast**  
#### Thanks for the response - my problem is that it seems nocoDB requires an "Id" field. I am using REST POST api from a source that can't send an incrementing "Id" field just sensor values. If I try to send the JSON formatted fields without an "Id" field I get a 400 Error. I guess I'm asking if there is a way for nocoDB to work out that there is no Id value and automatically add one to inputs coming from the POST api. I can add a timestamp field as "Id" but that doesn't seem to work. Any possible workarounds would be appreciated.  
#  
> 💁 **o1lab**  
#### can u pls confirm if you are using this api  
#  
> 💁 **o1lab**  
#### https://apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-create  
#  
> 💁 **o1lab**  
#### it works as you are proposing  
#  
> 💁 **o1lab**  
#### and wht are the error like on server side  
#  
> 🙋‍♂️ **geocoast**  
#### Hi, yes that is the api I am using. I was trying to follow the example schema from the api documents:Example Value   
#### Schema   
#### {   
####   "Id": 0,   
####   "CreatedAt": "string",   
####   "UpdatedAt": "string",   
####   "time": 0,   
####   "precipMax": 0   
#### }  
#  
> 🙋‍♂️ **geocoast**  
#### However when I removed the "Id" and created/updated fields from the JSON I was sending and just sent the values the "Id" was correctly autogenerated. Seems like by adding the created/updated fields I was preventing the "Id" from being autogenerated. I'm sure this might be obvious to others but perhaps the example schema should not include the autogenerated variables? The link you sent about the api didn't have these so perhaps they should be removed from the examples in the installed version NocoDB: API Docs. Again this may not be a problem for many users but may help inexperience ones like myself! Thanks again for helping to get me on the right path.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/986281262035648522)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
